spec.html

```diff
@@ -66,124 +66,400 @@ <h1><ins>AsyncBlockStart ( _promiseCapability_, _asyncBody_, _asyncContext_ )</i
+          1. <ins>If _requiredModule_.[[Status]] is `"evaluated"`, then</ins>
+            1. <ins>Add _requiredModule_.[[ExecPromise]] to the list _dependencyExecPromises_.</ins>
+          1. If _requiredModule_.[[Status]] is `"evaluating"`, then
+            1. Assert: _requiredModule_ is a Source Text Module Record.
+            1. Set _module_.[[DFSAncestorIndex]] to min(_module_.[[DFSAncestorIndex]], _requiredModule_.[[DFSAncestorIndex]]).
+            1. <ins>Let _stackErrorCapability_ be ! NewPromiseCapability(%Promise%).</ins>
+            1. <ins>Perform ! PerformPromiseThen(_requiredModule_.[[ExecPromise]], *undefined*, _stackErrorCapability_.[[Reject]]).</ins>
+            1. <ins>Add _stackErrorCapability_ to the list _dependencyExecPromises_.</ins>
+        1. <del>Perform ? ModuleExecution(_module_).</del>
+        1. <ins>Let _executionSteps_ be the algorithm steps defined in ModuleExecution.</ins>
+        1. <ins>Let _execution_ be CreateBuiltinFunction(_executionSteps_, &laquo; Module, DependencyExecPromises, Resolve, Reject &raquo;).</ins>
+        1. <ins>Set _execution_.[[Module]] to _module_.</ins>
+        1. <ins>Set _execution_.[[DependencyExecPromises]] to _dependencyExecPromises_.</ins>
+        1. <ins>Perform ! PerformBuiltinPromiseThen(_execution_, _evalCapability_.[[Resolve]], _evalCapability_.[[Reject]]).</ins>
+        1. Assert: _module_ occurs exactly once in _stack_.
+        1. Assert: _module_.[[DFSAncestorIndex]] is less than or equal to _module_.[[DFSIndex]].
+        1. If _module_.[[DFSAncestorIndex]] equals _module_.[[DFSIndex]], then
+          1. Let _done_ be *false*.
+          1. Repeat, while _done_ is *false*,
+            1. Let _requiredModule_ be the last element in _stack_.
+            1. Remove the last element of _stack_.
+            1. Set _requiredModule_.[[Status]] to `"evaluated"`.
+            1. If _requiredModule_ and _module_ are the same Module Record, set _done_ to *true*.
+        1. Return _index_.
+      </emu-alg>
+      <emu-note>
+        <p>A module is `"evaluating"` while it is being traversed by InnerModuleEvaluation. A module is `"evaluated"` when ModuleExecution has been called, even if that module execution is a promise that has not yet resolved.</p>
+      </emu-nnote>
+    </emu-clause>
+
+    <emu-clause id="sec-moduleexecution" aoid="ModuleExecution">
+      <h1>ModuleExecution( _module_, _dependencyExecPromises_, _resolve_, _reject_ )</h1>
+
+      <p>The ModuleExecution abstract operation is used by InnerModuleEvaluation to initialize the execution context of the module and evaluate the module's code within it.</p>
+
+      <p>This abstract operation performs the following steps:</p>
+
+      <emu-alg>
+        1. <ins>For each Promise _promise_ in _dependencyExecPromises_, do</ins>
+          1. <ins>AwaitWithReturn(_module_.[[ExecPromise]]).</ins>
```

Comment:
I'm not sure if this is the behavior we want to have on errors. If one module's Promise rejects, shouldn't we reject this module's Promise immediately, without waiting for the earlier ones to resolve?

Reply:
Note that the logic in https://github.com/w3ctag/promises-guide/pull/55 handles failures in this way.